### PR TITLE
LPS-192516 Cannot use entity fields as variable for Recipient field when separating variables by comma and space or semicolon

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
Issue : https://liferay.atlassian.net/browse/LPS-192516

### CODE REVIEW

**Comment by pedro-oliveira446** [comment]
ci:stop

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 1]
```diff
@@ -103,64 +108,68 @@ public void testSendNotification() throws Exception {
...
+
+		_testSendNotificationWithToProperty(
+			expectedToEmailAddresses2,
+			"[%CURRENT_USER_EMAIL_ADDRESS%], " +
+				getTermName("emailTextObjectField"));
+
+		_testSendNotificationWithToProperty(
+			expectedToEmailAddresses2,
+			"[%CURRENT_USER_EMAIL_ADDRESS%];" +
+				getTermName("emailTextObjectField"));
```
Comment (p-albuquerque) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:153]: I believe we can make this part more short, and inline some variables attribution:

```java
		for (String separator :
				Arrays.asList(
					StringPool.COMMA, StringPool.COMMA_AND_SPACE,
					StringPool.SEMICOLON)) {

			_testSendNotification(
				ListUtil.sort(
					Arrays.asList(
						user1.getEmailAddress(), user2.getEmailAddress())),
				StringBundler.concat(
					user1.getEmailAddress(), separator,
					user2.getEmailAddress()));

			_testSendNotification(
				ListUtil.sort(
					Arrays.asList(
						user2.getEmailAddress(),
						GetterUtil.getString(
							childObjectEntryValues.get(
								"emailTextObjectField")))),
				StringBundler.concat(
					"[%CURRENT_USER_EMAIL_ADDRESS%]", separator,
					getTermName("emailTextObjectField")));
		}
```

**Review by p-albuquerque — CHANGES_REQUESTED** [review]

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Review by p-albuquerque — APPROVED** [review]
LGTM! :100:

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
Hey @pedro-oliveira446 please take a look at [this pull request](https://github.com/MarinhoFeliphe/liferay-portal/pull/234) and try to apply the same pattern used for tests.

Nice changes by the way! :tada: 

Thank you!

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 2]
```diff
@@ -102,64 +84,90 @@ public void testSendNotification() throws Exception {
 		notificationQueueEntryLocalService.deleteNotificationQueueEntry(
 			notificationQueueEntries.get(0));
 
-		notificationTemplate = _addNotificationTemplate(true);
+		// Multiples emails for each main recipients, comma separator
```
Comment (MarinhoFeliphe) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:87]: @pedro-oliveira446 just:

```suggestion
		// Multiples emails for each main recipient, comma separator
```

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 3]
```diff
@@ -265,4 +242,89 @@ private void _assertNotificationQueueEntry(
...
+					).build();
+				}
+			},
+			ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		objectActionLocalService.deleteObjectAction(
+			objectAction.getObjectActionId());
+	}
+
+	private void _testSendNotification(List<String> expectedTo, String to)
```
Comment (MarinhoFeliphe) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:291]: @pedro-oliveira446 rename this way:

```suggestion
	private void _testSendNotification(List<String> expectedToEmailAdresses, String to)
```

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 4]
```diff
@@ -58,30 +60,10 @@ public class EmailNotificationTypeTest extends BaseNotificationTypeTest {
...
 	@Test
 	public void testSendNotification() throws Exception {
-		Assert.assertEquals(
-			0,
-			notificationQueueEntryLocalService.
-				getNotificationQueueEntriesCount());
 
-		NotificationTemplate notificationTemplate = _addNotificationTemplate(
-			false);
+		// One email including all main recipients
```
Comment (MarinhoFeliphe) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:64]: @pedro-oliveira446 sort these blocks, the block:

// One email including all main recipients

should be after:

// Multiples emails for each main recipients, with terms and semicolon separator

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 5]
```diff
@@ -58,30 +60,10 @@ public class EmailNotificationTypeTest extends BaseNotificationTypeTest {
...
-			ObjectActionExecutorConstants.KEY_NOTIFICATION,
-			ObjectActionTriggerConstants.KEY_ON_AFTER_ADD,
-			UnicodePropertiesBuilder.put(
-				"notificationTemplateId",
-				notificationTemplate.getNotificationTemplateId()
-			).build(),
-			false);
-
-		_addObjectEntry();
+		_executeNotificationObjectAction(_addNotificationTemplate(false));
```
Comment (MarinhoFeliphe) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:66]: @pedro-oliveira446 try to apply the method `_testSendNotification` here as well,

Thanks!

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 6]
```diff
@@ -102,64 +84,90 @@ public void testSendNotification() throws Exception {
...
+		// separator
+
+		_testSendNotification(
+			ListUtil.sort(
+				Arrays.asList(
+					user2.getEmailAddress(),
+					GetterUtil.getString(
+						childObjectEntryValues.get("emailTextObjectField")))),
+			"[%CURRENT_USER_EMAIL_ADDRESS%];" +
+				getTermName("emailTextObjectField"));
```
Comment (MarinhoFeliphe) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:151]: @pedro-oliveira446 maybe we can do only one test for checking separators, for example:

```java
StringBundler.concat(
				user1.getEmailAddress(), StringPool.COMMA,
				user2.getEmailAddress(), StringPool.COMMA_AND_SPACE,
				"[%CURRENT_USER_EMAIL_ADDRESS%];",
				user3.getEmailAddress())
```

Comment (MarinhoFeliphe) [reply modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:151]: but I'm not sure, what do you think?

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java (hunk 6 above)
Comment (pedro-oliveira446) [reply modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:140]: @MarinhoFeliphe,
I think the way it is now, it's clearer what we're testing and it's the same way as requested in the bug. But, I'm not sure about that either.

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 7]
```diff
@@ -58,106 +60,102 @@ public class EmailNotificationTypeTest extends BaseNotificationTypeTest {
...
+				Arrays.asList(
+					user1.getEmailAddress(), user2.getEmailAddress())),
+			true,
+			StringBundler.concat(
+				user1.getEmailAddress(), StringPool.SEMICOLON,
+				user2.getEmailAddress()));
 
-		for (NotificationQueueEntry notificationQueueEntry :
-				notificationQueueEntries) {
+		// Multiples emails for each main recipient, with terms and comma
```
Comment (MarinhoFeliphe) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:100]: @pedro-oliveira446 just:

```suggestion
		// Multiples emails for each main recipient, terms, comma
```

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Review by MarinhoFeliphe — APPROVED** [review]
LGTM!! :tada:

**Comment by pedro-oliveira446** [comment]
ci:forward:force

**Comment by pedro-oliveira446** [comment]
ci:forward:force

**Pull request closed without merging**